src/shell/dotnet/Shell/Shell.csproj

```diff
@@ -47,16 +47,31 @@
 	</ItemGroup>
 
 	<PropertyGroup>
-		<Fdc3BundleOutput>$(ComposeUIRepositoryRoot)/src/shell/js/composeui-fdc3/dist/fdc3-iife-bundle.js</Fdc3BundleOutput>
+		<Fdc3BundleOutput>$(ComposeUIRepositoryRoot)\src\shell\js\composeui-fdc3\dist\fdc3-iife-bundle.js</Fdc3BundleOutput>
 	</PropertyGroup>
 
 	<ItemGroup>
-		<EmbeddedResource Include="$(Fdc3BundleOutput)">
-		</EmbeddedResource>
+		<EmbeddedResource Include="$(Fdc3BundleOutput)" />
 	</ItemGroup>
 
-	<Target Name="PreBuild" BeforeTargets="PrebuildEvent" Condition="!Exists('$(Fdc3BundleOutput)')">
-		<Exec Command="cd $(ComposeUIRepositoryRoot)" />
-		<Exec Command="npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
+	<Target Name="InstallMonoRepoDependencies"
+		Condition="!Exists('$(ComposeUIRepositoryRoot)\node_modules')">
+		<Message Importance="high" Text="Installing node_modules..." />
+		<Exec Command="cd $(ComposeUIRepositoryRoot) &amp; npm install" />
+		<Message Importance="normal" Text="Finished command: npm install." />
 	</Target>
+
+	<Target Name="RunFdc3LernaBuild">
+		<Message Importance="high" Text="Running npx lerna run build for composeui-fdc3..." />
+		<Exec Command="cd $(ComposeUIRepositoryRoot) &amp; npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
+		<Message Importance="normal" Text="Finished building the fdc3 bundle." />
+	</Target>
+
+	<Target
```

Comment:
I'd add back the condition, otherwise this will run with every build, even if the bundle exists. 
Maybe even remove the separate RunFdc3LernaBuild target, as it is not triggered from anywhere else.
```xml
    <Target Name="BuildComposeUIFdc3Bundle" BeforeTargets="PrebuildEvent">        
        <CallTarget Targets="InstallMonoRepoDependencies" />
        <Message Importance="high" Text="Running npx lerna run build for composeui-fdc3..." />
        <Exec Command="cd $(ComposeUIRepositoryRoot) &amp;&amp; npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
        <Message Importance="normal" Text="Finished building the fdc3 bundle." />
    </Target>
```

Reply:
I managed to leave the condition out of the target...
```
    <Target Name="BuildComposeUIFdc3Bundle" BeforeTargets="PrebuildEvent" Condition="!Exists('$(Fdc3BundleOutput)')">        
        <CallTarget Targets="InstallMonoRepoDependencies" />
        <Message Importance="high" Text="Running npx lerna run build for composeui-fdc3..." />
        <Exec Command="cd $(ComposeUIRepositoryRoot) &amp;&amp; npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
        <Message Importance="normal" Text="Finished building the fdc3 bundle." />
    </Target>
```